api/src/main/java/com/epam/pipeline/manager/docker/scan/AggregatingToolScanManager.java

```diff
@@ -235,6 +237,16 @@ public boolean checkTool(Tool tool, String tag) {
                     maxMediumVulnerabilities < severityCounters.getOrDefault(VulnerabilitySeverity.Medium, 0)) {
                 return false;
             }
+
+            LOGGER.debug("Tool: " + tool.getId() + " version: " + tag +
+                    "Check tool os version.");
+            if (toolVersionScanResult.getToolOSVersion() != null
+                    && !toolVersionScanResult.getToolOSVersion().getIsAllowed()) {
```

Comment:
I think we can get `SystemPreference` value and check os version at this point and remove `isAllowed` field from object itself.